Lib/idlelib/query.py

```diff
@@ -302,10 +303,61 @@ def entry_ok(self):
         path = self.path_ok()
         return None if name is None or path is None else (name, path)
 
+class CustomRun(Query):
+    "Get arguments for custom run module."
+    # Used in runscript.run_custom_event
+
+    def __init__(self, parent, title, *, cli_args='',
+                 _htest=False, _utest=False):
+        """Get command line args for custom run environment.
+
+        User enters the command line arugments for running the file.
+        """
+        message = 'Command Line Arguments:'
+        super().__init__(
+                parent, title, message, text0=cli_args,
+                _htest=_htest, _utest=_utest)
+
+    def create_widgets(self):
+        super().create_widgets(ok_text='Run')
+        frame = self.frame
+        self.restartvar = BooleanVar(self, value=True)
+        restart = Checkbutton(frame, variable=self.restartvar, onvalue=True,
+                              offvalue=False, text='Restart shell')
+        self.args_error = Label(frame, text=' ', foreground='red',
+                                font=self.error_font)
+
+        restart.grid(column=0, row=4, columnspan=3, padx=5, sticky='w')
+        self.args_error.grid(column=0, row=12, columnspan=3, padx=5,
+                             sticky='we')
```

Comment:
I'm not using `self.args_error` yet, but it added nice spacing and I figured it would come in handy for an edit such as `must restart with arguments` as mentioned on the bug tracker.

Reply:
Yes, `restart` works now, but it's never disabled.  I wasn't sure which edits would make sense around it.  On the ticket, you had mentioned that using CLI arguments would make sense on a restart, but right now it allows changing the CLI arguments without restarting the shell.  That's where my other comments (about this dialog not allowing blank arguments) and what the current `Run Module`'s behavior should be.